# HG changeset patch
# User John Donoghue
# Date 1540392556 14400
#      Wed Oct 24 10:49:16 2018 -0400
# Node ID 12ad5eb2328efc1aab693d90dea3f9368fd0a3a3
# Parent  f7b205562f1dcb0089b3fb034464822564ced575
Call the document browser copy on clipboard signal (Bug #54879)

* libgui/src/documentation.cc
  (documentation::copyClipboard): call m_doc_browser->copy() if have focus

diff --git a/libgui/src/documentation.cc b/libgui/src/documentation.cc
--- a/libgui/src/documentation.cc
+++ b/libgui/src/documentation.cc
@@ -369,17 +369,23 @@ namespace octave
 
   void documentation::notice_settings (const QSettings *)
   {
     shortcut_manager::shortcut (m_show_shortcut, "editor_edit:find_replace");
     shortcut_manager::shortcut (m_findnext_shortcut, "editor_edit:find_next");
     shortcut_manager::shortcut (m_findprev_shortcut, "editor_edit:find_previous");
   }
 
-  void documentation::copyClipboard (void) { }
+  void documentation::copyClipboard (void)
+  {
+    if (m_doc_browser->hasFocus ())
+      {
+        m_doc_browser->copy();
+      }
+  }
 
   void documentation::pasteClipboard (void) { }
 
   void documentation::selectAll (void) { }
 
   void documentation::load_ref (const QString& ref_name)
   {
     if (! m_help_engine)
